### PR TITLE
US012 - Add a new role

### PR DESCRIPTION
**Pull request by ManpreetM01**

### CODE REVIEW

File: src/main/java/org/kainos/ea/db/JobRoleDao.java [hunk 1]
```diff
@@ -82,4 +79,35 @@ public int updateJobRole(int id, UpdateJobRoleRequest jobRoleRequest) throws SQL
             throw new FailedToUpdateJobRoleException();
         }
     }
+
+    public int createJobRole(CreateJobRoleRequest jobRoleRequest) throws SQLException, FailedToCreateJobRoleRequestException {
+        Connection c = databaseConnector.getConnection();
+
+        try {
+            String insertJobRoleQuery = "INSERT INTO JobRoles (jobRoleName, jobSpecSummary, bandID, capabilityID, responsibilities, sharePoint) " +
```
Comment (TomBillingtonUK) [inline src/main/java/org/kainos/ea/db/JobRoleDao.java:113]: Is there a way you can make sure if one of the foreign keys is invalid that you return a 400 Bad Request to the user

File: src/test/java/org/kainos/ea/dao/JobRoleDaoTest.java [hunk 2]
```diff
@@ -125,4 +126,44 @@ public void updateJobRole_shouldThrowFailedToUpdateJobRoleException_whenCantUpda
...
+
+        Connection connectionMock = mock(Connection.class);
+        Mockito.when(databaseConnector.getConnection()).thenThrow(SQLException.class);
+        PreparedStatement mockPreparedStatement = mock(PreparedStatement.class);
+        Mockito.when(connectionMock.prepareStatement(any(String.class))).thenReturn(mockPreparedStatement);
+        Mockito.when(mockPreparedStatement.executeUpdate()).thenReturn(-1);
+        assertThrows(SQLException.class,
+                () -> jobRoleDao.createJobRole(jobRoleRequest));
+    }
 }
```
Comment (TomBillingtonUK) [inline src/test/java/org/kainos/ea/dao/JobRoleDaoTest.java]: Missing test case when -1 is returned from the DAO

File: src/test/java/org/kainos/ea/service/JobRoleServiceTest.java [hunk 3]
```diff
@@ -101,4 +110,30 @@ public void updateJobRole_shouldReturnID_whenSuccess() throws SQLException, Fail
...
+        assertThrows(FailedToCreateJobRoleRequestException.class, () -> jobRoleService.createJobRole(createJobRoleRequest));
+    }
+
+    @Test
+    public void createJobRole_shouldThrowInvalidJobRoleException_whenInvalidJobRoleRequest() throws InvalidJobRoleException {
+        Mockito.when(createJobRoleValidator.isValidJobRole(any(CreateJobRoleRequest.class))).thenReturn("Test invalid job role");
+        assertThrows(InvalidJobRoleException.class,
+                () -> jobRoleService.createJobRole(createJobRoleRequest));
+    }
 }
```
Comment (TomBillingtonUK) [inline src/test/java/org/kainos/ea/service/JobRoleServiceTest.java:254]: Missing the test case where DAO returns -1

File: src/test/java/org/kainos/ea/dao/JobRoleDaoTest.java (hunk 2 above)
Comment (ManpreetM01) [reply src/test/java/org/kainos/ea/dao/JobRoleDaoTest.java]: added

File: src/test/java/org/kainos/ea/service/JobRoleServiceTest.java (hunk 3 above)
Comment (ManpreetM01) [reply src/test/java/org/kainos/ea/service/JobRoleServiceTest.java:254]: added

File: src/main/java/org/kainos/ea/db/JobRoleDao.java (hunk 1 above)
Comment (ManpreetM01) [reply src/main/java/org/kainos/ea/db/JobRoleDao.java:113]: changes made

**Review by TomBillingtonUK — APPROVED** [review]

**Review by RobertCrabtreeKainos — APPROVED** [review]

**Pull request merged**